x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/field_config.tsx

```diff
@@ -1251,4 +1281,168 @@ export const FIELD = (readOnly?: boolean): FieldMap => ({
+      defaultMessage: 'Response body max bytes',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.responseBodyMaxBytes.helpText', {
+      defaultMessage: 'Control the maximum size of the stored body contents.',
+    }),
+    hidden: (dependencies) => {
+      const [responseBodyIndex] = dependencies || [];
+      return responseBodyIndex === ResponseBodyIndexPolicy.NEVER;
+    },
+    props: (): EuiFieldNumberProps => ({ min: 1, step: 'any', readOnly }),
+    dependencies: [ConfigKey.RESPONSE_BODY_INDEX],
+  },
+  [ConfigKey.IPV4]: {
+    fieldKey: ConfigKey.IPV4, // also controls ipv6
+    component: ComboBox,
+    label: i18n.translate('xpack.synthetics.monitorConfig.ipv4.label', {
+      defaultMessage: 'IP protocols',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.ipv4.helpText', {
+      defaultMessage: 'Specifies which IP procols to use when pinging the remote host.',
+    }),
+    controlled: true,
+    dependencies: [ConfigKey.IPV6],
+    props: ({ field, setValue, dependencies }): EuiComboBoxProps<string> => {
+      const [ipv6] = dependencies;
+      const ipv4 = field?.value;
+      const values: string[] = [];
+      if (ipv4) {
+        values.push('IPv4');
+      }
+      if (ipv6) {
+        values.push('IPv6');
+      }
+      return {
+        options: [
+          {
+            label: 'IPv4',
+          },
+          {
+            label: 'IPv6',
+          },
+        ],
+        selectedOptions: values.map((version) => ({
+          label: version,
+        })),
+        onChange: (updatedValues: Array<EuiComboBoxOptionOption<string>>) => {
+          setValue(
+            ConfigKey.IPV4,
+            updatedValues.some((value) => value.label === 'IPv4')
+          );
+          setValue(
+            ConfigKey.IPV6,
+            updatedValues.some((value) => value.label === 'IPv6')
+          );
+        },
+        isDisabled: readOnly,
+      };
+    },
+  },
+  [ConfigKey.PROXY_HEADERS]: {
+    fieldKey: ConfigKey.PROXY_HEADERS,
+    component: HeaderField,
+    label: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.label', {
+      defaultMessage: 'Proxy headers',
+    }),
+    helpText: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.helpText', {
+      defaultMessage: 'Additional headers to send to proxies during CONNECT requests.',
+    }),
+    controlled: true,
+    validation: () => ({
+      validate: (headers) => !validateHeaders(headers),
+    }),
+    error: i18n.translate('xpack.synthetics.monitorConfig.proxyHeaders.error', {
+      defaultMessage: 'Header key must be a valid HTTP token.',
+    }),
+    props: (): HeaderFieldProps => ({
+      readOnly,
+    }),
+  },
+  ['check.response.json']: {
+    fieldKey: ConfigKey.RESPONSE_JSON_CHECK,
+    component: KeyValuePairsField,
+    label: i18n.translate('xpack.synthetics.monitorConfig.responseJSON.label', {
+      defaultMessage: 'Check response body contains JSON',
```

Comment:
I noticed that all labels for similar settings are like this. I'd suggest 2 improvements depending on what is possible, timely and realistic to do:

1. Have simple labels and use the operators as preset fields (a bit like what the JSON setting looks like), for example:
**Response status**
`Equals to`        `input field...`
**Response body**
`Contains`    `input field...`
_some help text_
`Does not contain`        `input field...`
 _some help text_
 
2. Quick label structure improvement: remove "Check". It's already explained in the section description and name "Response checks"
- Response body contains JSON
- Response status equals to
- Response headers contain
- etc.